src/kv/RocksDBStore.cc

```diff
@@ -2194,10 +2229,29 @@ class CFIteratorImpl : public KeyValueDB::IteratorImpl {
 protected:
   string prefix;
   rocksdb::Iterator *dbiter;
+  const KeyValueDB::IteratorBounds bounds;
+  const rocksdb::Slice iterate_lower_bound;
+  const rocksdb::Slice iterate_upper_bound;
 public:
-  explicit CFIteratorImpl(const std::string& p,
-				 rocksdb::Iterator *iter)
-    : prefix(p), dbiter(iter) { }
+  explicit CFIteratorImpl(const RocksDBStore* db,
+                          const std::string& p,
+                          rocksdb::ColumnFamilyHandle* cf,
+                          KeyValueDB::IteratorBounds bounds_)
+    : prefix(p), bounds(std::move(bounds_)),
+      iterate_lower_bound(make_slice(bounds.lower_bound)),
+      iterate_upper_bound(make_slice(bounds.upper_bound))
+      {
+      auto options = rocksdb::ReadOptions();
+      if (db->cct->_conf.get_val<bool>("osd_rocksdb_iterator_bounds_enabled")) {
```

Comment:
`get_val()` unfortunately requires a global mutex that we probably don't want to pay for here. if you add this line to the config option:
```
with_legacy: true
```
you can access this configurable directly without locking:
```c++
if (db->cct->_conf->osd_rocksdb_iterator_bounds_enabled) {
```

Reply:
thanks @cbodley, I was wondering what affect "with_legacy" had. Updating it now!